autodevtech/autodevstats
# split statstool into 2 smaller scripts and fix a bug with ordering of API results

This is part of 4 PRs: autodevstats, datascience, and labor, and ops all need to be updated.

The plan to roll this out is to:
1. push autodevstats
1. upload the tarball to s3
1. validate everything works (use --create distribution to use new analyze_repo)
1. push datascience
1. update the commit that labor uses for datascience and ops for 
1. push labor
1. push ops
1. deploy with ansible

Steps to verify things are working:
* Ran statstool (pre-changes) on random repo (https://github.com/terkelg/zet)
  * with SKIP_ANALYSIS set to false (models old behavior)
* Ran statstool && fetch_github_data
  * validated equivalent output as before
  
Some more context to help digest these changes:
* The bug is: sort=desc --> direction=desc
* Deleted everything related to analysis (there are 2 entry points to this script, both of which set skip_analysis = True)
* removed the conditional check for DATADIR (both entry points pass a DATADIR so it should never be empty)
* removed the code contained by "if -d DATADIR" (duplicate code of the first part of fetch_github_data, that never gets executed because the datadir is created by the top part of statstool, not the caller)
* moved the generation of the timestamp for data_pull_start_time to the caller (it's shared between statstool and fetch_github_data)
